src/components/schedule-visualizer/index.tsx

```diff
@@ -214,9 +216,35 @@ export default function ScheduleVisualizer(props: ScheduleVisualizerProps): Reac
     return () => clearInterval(interval);
   }, [props.trajManager, curMapFloorLayer, trajAnimDuration]);
 
+  function determineMinZoom(width: number, height: number): number {
+    if (width >= height * -1) {
```

Comment:
It must be a reason for you to multiply it for` -1`, but is there a way where you can compare the absolute values?
Something like:
Math.abs(width)>=Math.abs(height)